src/Browser/LoginButton.js

```diff
@@ -1,108 +1,199 @@
-const HumanInteractionInterface = require('../Auth/HumanInteractionInterface');
+const Cookies = require('./CookieUtils');
 const AuthStates = require('../Auth/AuthStates');
+const AuthController = require('../Auth/AuthController');
+const Service = require('../Service');
+const Messages = require('../Auth/LoginMessages');
+const utils = require('../utils');
 
 /**
  * @memberof Pryv.Browser
  */
-class LoginButton extends HumanInteractionInterface {
+class LoginButton {
 
-  /**
-   * @param {Browser/AuthController} auth 
-   */
-  
-  constructor (authController) {
-    super(authController);
+  constructor(authSettings, service) {
+    this.authSettings = authSettings;
+    this.service = service;
+    this.serviceInfo = service.infoSync();
   }
-   
+
   /**
    * setup button and load assets
    */
   async init () {
-    this.auth.store.stateChangeListners.push(this.onStateChange.bind(this));
-    this._setupButton();
-    await this._loadAssets();   
+    // initialize button visuals
+    setupButton(this);
+    this.languageCode = this.authSettings.authRequest.languageCode || 'en';    
+    this.messages = Messages(this.languageCode);
+    if (this.loginButtonText) {
+      await loadAssets(this);
+    }
+    // set cookie key for authorization data
+    this._cookieKey = 'pryv-libjs-' + this.authSettings.authRequest.requestingAppId;
+    
+    // initialize controller
+    if (this.auth) { return this.auth; }
+    this.auth = new AuthController(this.authSettings, this.service, this);
+    await this.auth.init();
+
+    return this.service;
+  }
+
+  onClick () {
+    this.auth.handleClick();
   }
-  
-  _setupButton () {
-    this.loginButtonSpan = document.getElementById(this.auth.settings.spanButtonID);
 
-    if (!this.loginButtonSpan) {
-      console.log('WARNING: Pryv.Browser initialized with no spanButtonID');
+  async onStateChange (state) {
+    switch (state.status) {
+      case AuthStates.LOADING:
+        this.text = getLoadingMessage(this);
+        break;
+      case AuthStates.INITIALIZED:
+        this.text = getInitializedMessage(this, this.serviceInfo.name);
+        break;
+      case AuthStates.NEED_SIGNIN:
+        const loginUrl = state.authUrl || state.url; // url is deprecated
+        if (this.authSettings.authRequest.returnURL) { // open on same page (no Popup) 
+          location.href = loginUrl; 
+          return;
+        } else {
+          startLoginScreen(this, loginUrl);
+        }
+        break;
+      case AuthStates.AUTHORIZED:
+        this.text = state.username;
+        this.saveAuthorizationData({
+          apiEndpoint: state.apiEndpoint,
+          username: state.username
+        });
+        break;
+      case AuthStates.LOGOUT:
+        const message = this.messages.LOGOUT_CONFIRM ? this.messages.LOGOUT_CONFIRM : 'Logout ?';
+        if (confirm(message)) {
+          this.deleteAuthorizationData();
```

Comment:
Not sure if it was so or is it a new behariour but for the login with redirection prYvpoll should be cleaned from the url, otherwise it is impossible to logout.

Reply:
I saw indeed, but it's not handled today - so I put it out of scope:
See https://api.pryv.com/app-web-access/?pryvServiceInfoUrl=https://reg.pryv.me/service/info
show advanced options, return url to self#